src/Analyzers/CSharp/Tests/UseCollectionExpression/UseCollectionExpressionForArrayTests.cs

```diff
@@ -4409,6 +4409,43 @@ IEnumerable<int> M()
         }.RunAsync();
     }
 
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/70996")]
+    public async Task TestInterfaceOn_ReadWriteDestination()
+    {
+        await new VerifyCS.Test
+        {
+            TestCode =
+                """
+                using System;
+                using System.Collections.Generic;
+                using System.Linq.Expressions;
+
+                class C
+                {
+                    IList<int> M()
+                    {
+                        return [|[|new|] int[]|] { 1, 2, 3 };
```

Comment:
this case is fine to convert as `new int[]...` gives you a new instance.  so it's allocating beore and after.